sql: fix ADD COLUMN ... UNIQUE for PARTITION ALL BY tables

otan opened the pull request:
Refs: #63113

Release note (bug fix): Fixed a bug where doing ALTER TABLE ... ADD
COLUMN ... UNIQUE would error if the table had a PARTITION ALL BY /
REGIONAL BY ROW definition.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63123)
<!-- Reviewable:end -->

otan committed:
sql: fix ADD COLUMN ... UNIQUE for PARTITION ALL BY tables

Release note (bug fix): Fixed a bug where doing ALTER TABLE ... ADD
COLUMN ... UNIQUE would error if the table had a PARTITION ALL BY /
REGIONAL BY ROW definition.

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63123#-:-MXdNf7n40c2Iyv2tLWC:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
